### ISSUE TITLE
Stops recording to storage device

### ISSUE THREAD

**Issue by wosss** [waynieack/HikvisionZM#6]
Hi, after some random time it stops recording to storage device even the plugin keeps sending VMD events. This problem seems to appear in version 1.36.x and forward, at 1.34.x it was working perfectly. Killing motionstream's PID and starting again is enough to record again but after some random time the problem repeats. 

As additional data:
Raspbian, ZM 1.34 = works ok.
CentOS 7, ZM 1.36 = stops recording.
Fedora 33, ZM 1.37 = stops recording.

**Comment by waynieack** [comment]
I assume your talking about /dev/shm filling up and zm stops recording.

Are you running the latest master version?

If you are running the latest, I may have not pushed the fixed version. I
did finally figure out this issue because I was also having it in the newer
ZM versions which I worked around by adding quite a bit of memory.

**Comment by wosss** [comment]
Regarding /dev/shm, it also stopped in a HPe Server with 32GB available in shm. I have connected 4 cameras recording at 1080p@15fps. 

Yes I'm using latest master, "grep Version" reports 1.2.3. One thing I noticed, it doesn't stop recording all 4 cameras at once, it keeps stopping between several hours until last camera stop with no order apparently. Also manual alarm trigger in ZM web interface keeps working and recording even when plugin stops recording.

**Comment by waynieack** [comment]
Yeah, more memory + increasing the shm size. I'll get the fixed version
pushed when I get a chance. I am out of town right now but I should be able
to get to it in the next day or two.

**Comment by waynieack** [comment]
I just pushed the fixed version, give it a try and let me know.

**Comment by wosss** [comment]
Thank you, tested the fixed version about 4 days. Today one of 4 cameras stopped recording for 2 hours and then it started recording again, the other 3 are fine. I'll keep testing a few days and report back.

**Comment by waynieack** [comment]
Did /dev/shm fill up?

Do you have the script logs for when the issue occurred?

**Comment by wosss** [comment]
No, /dev/shm usage keeps around 200MB from 32GB available, using 4 cameras 1920x1080@15 fps

Logs are empty, but no issues appeared on 6 days since last post.
